docs/src/modules/components/ThemeContext.js

```diff
@@ -65,14 +65,6 @@ const highDensity = {
       defaultProps: {
         size: 'small',
       },
-      styleOverrides: {
-        sizeSmall: {
-          // minimal touch target hit spacing
-          marginLeft: 4,
-          marginRight: 4,
-          padding: 12,
-        },
-      },
```

Comment:
This makes the result super confusing. `padding: 12` is size large.

Reply:
For whom "super confusing"? Did you consider the code comment?

Reply:
The user who try the playground would be confused because by toggling the dense switch, they expect to be smaller but the `IconButton` is bigger.

Reply:
https://user-images.githubusercontent.com/18292247/135822234-eeff340e-8180-43e4-872b-793fe1ef6a28.mov

Reply:
And how does it look now? Did you ensure the code comment still applies?

Reply:
This is what it looks now. 

https://user-images.githubusercontent.com/18292247/135827069-09316715-3c95-4244-ae25-15842722971a.mov

What do you think about removing the `styleOverride: ` part, or do you think we should keep it?

Reply:
@oliviertassinari any thought?